### ISSUE TITLE
Default configuration of Notepad++ breaks Python files by adding tabs instead of spaces

### ISSUE THREAD

**Issue by WoLpH** [notepad-plus-plus/notepad-plus-plus#5506]
### Description of the Issue

Note: copy of #703

By default Python files should use spaces instead of tabs (https://www.python.org/dev/peps/pep-0008/#tabs-or-spaces) and with Python 3 the mixing of tabs and spaces is completely forbidden.

The result is that with the default of tab based indention in notepad++ after adding just a single line of code you break any existing files. So at least for Python the default should be "replace tabs with spaces" and "4 spaces per tab".

### Steps to Reproduce the Issue

1. Open a regular Python file that uses space indenting (as ~99% of the Python files do)
2. Add an indented line
3. The Python file won't work anymore

```python
  File "test.py", line 5
    print('test')
                  ^
TabError: inconsistent use of tabs and spaces in indentation
```

### Expected Behavior

Either default to spaces instead of tabs for Python files or automatically detect the setting from other indentations in the file.

### Actual Behavior

Always defaults to tabs when indenting.

**Comment by Ekopalypse** [comment]
Use Settings->Preferences->Language and either overwrite the global tab settings or choose
python and set it individually. As python allows both, tab and spaces, to be used there is no reason why to favor one over the other.

**Comment by WoLpH** [comment]
@Ekopalypse  Yes there is a reason to favour spaces. As you can read in the provided link, the official Python standard says spaces should be used. So 99% of the Python projects use spaces instead of tabs.

**Comment by Ekopalypse** [comment]
I agree, pep8 guideline states that spaces are the preferred ones and maybe most of the python project uses it but I would argue that most of the python programmers using npp are aware of it already and have changed, or not, the settings to their needs. 
Changing it now per default would only bring those in trouble who are using tabs. 
Personally, I don't think it is worth the effort but if the developer decide to do so I don't have any hard feelings.

**Comment by WoLpH** [comment]
While that's true, I think there's a solution that won't hurt anyone and will fix the issue for new users. Simply detect whether tabs or spaces are used at the first point of indenting. It's easy enough and at the very least it won't break files for anyone simply editing a Python file.

**Comment by WoLpH** [comment]
Also, changing the default would only change it for new users, right? Not for existing users that already have npp installed and configured. So backwards compatibility shouldn't be an issue here.

**Comment by Ekopalypse** [comment]
>Also, changing the default would only change it for new users, right?
Depends on how one installs a new version. An update would, most likely, keep the settings but a deinstallation, wiping the data and reinstalling wouldn't. 
I think the approach of scanning the first indented line is the only way to ensure that no one is going
to get surprised. 
The question that remains is, is this something the devs would like to implement as it would only be useful for python like languages.

**Comment by sasumner** [comment]
>I think the approach of scanning the first indented line is the only way to ensure that no one is going
to get surprised.

Maybe the only way to never be surprised is to have visible whitespace turned on. :)

**Comment by WoLpH** [comment]
If you wipe your settings you shouldn't expect your settings to remain :P

Regardless, the official standard for Python is spaces so I would argue that if you're surprised by it you explicitly chose it for your project. Most users would actually be surprised that editing a file with npp breaks the file immediately. A rather crappy experience if you ask me ;)

**Comment by Yay295** [comment]
> the official standard for Python is spaces

And yet Python 3 still supports using tabs instead. How strange.

That said I wouldn't be opposed to Python files defaulting to spaces with a fresh install, since it does seem to be the most common indentation for Python.

**Comment by WoLpH** [comment]
> And yet Python 3 still supports using tabs instead. How strange.

Some languages such as C support goto statements. Doesn't mean it's a good idea to use them though ;)

**Comment by Ekopalypse** [comment]
@donho, would you consider a change in langs.xml addressing this request?
`<Language name="python" ext="py pyw" commentLine="#" tabSettings="132">`
As yaml already has such an entry per default it doesn't seem to be a python only hack.

**Comment by donho** [comment]
@Ekopalypse Please remind me what 132 means?

**Comment by MetaChuh** [comment]
@donho 
`tabSettings` in `langs.xml`
132 == tab size 4 + 128 for `replace by space`

**Comment by WoLpH** [comment]
Thanks for the fix @donho :)

**Comment by KOLANICH** [comment]
>By default Python files should use spaces instead of tabs (https://www.python.org/dev/peps/pep-0008/#tabs-or-spaces)

No, they shouldn't. PEP-8 is just a badly-designed holly cow of python community.

>The result is that with the default of tab based indention in notepad++ after adding just a single line of code you break any existing files. 

Just install an `.editorconfig` plugin and add a file configuring indentation according to pep-8 into `Lib`.

>Either default to spaces instead of tabs for Python files or automatically detect the setting from other indentations in the file.

I vote against it. This should be configured, not hardcoded.

**Comment by WoLpH** [comment]
> > By default Python files should use spaces instead of tabs (https://www.python.org/dev/peps/pep-0008/#tabs-or-spaces)
> 
> No, they shouldn't. PEP-8 is just a badly-designed holly cow of python community.

You don't have to agree with it, but it is the de facto standard.

> > The result is that with the default of tab based indention in notepad++ after adding just a single line of code you break any existing files.
> 
> Just install an `.editorconfig` plugin and add a file configuring indentation according to pep-8 into `Lib`.

You are free to do that if you disagree with the standard of course :)

> > Either default to spaces instead of tabs for Python files or automatically detect the setting from other indentations in the file.
> 
> I vote against it. This should be configured, not hardcoded.

Noted. But your opinion does't (or at least shouldn't) have more weight than the current standard.

I believe XKCD is relevant here: https://xkcd.com/927/
![image](https://user-images.githubusercontent.com/270571/58584166-fd005800-8255-11e9-9ef3-5c24dcf43c07.png)

**Comment by sasumner** [comment]
@KOLANICH You're too late. The author has already agreed with 4-spaces.  I don't see this being revisited.
